src/WebJobs.Extensions.DurableTask/ContextImplementations/RemoteEntityContext.cs

```diff
@@ -20,9 +19,40 @@ public RemoteEntityContext(EntityBatchRequest batchRequest)
         }
 
         [JsonProperty("request")]
-        public EntityBatchRequest Request { get; private set; }
+        internal EntityBatchRequest Request { get; private set; }
 
         [JsonIgnore]
         internal EntityBatchResult? Result { get; set; }
+
+        internal void EnsureSuccess()
+        {
+            if (this.Result == null)
+            {
+                throw new InvalidOperationException("Entity batch request has not been processed yet.");
+            }
+
+            if (this.Result.FailureDetails is { } f)
+            {
+                throw new EntityFailureException(f.ErrorMessage);
+            }
+
+            List<Exception>? errors = null;
+            if (this.Result.Results is not null)
+            {
+                foreach (OperationResult result in this.Result.Results)
+                {
+                    if (result.FailureDetails is { } failure)
+                    {
+                        errors ??= new List<Exception>();
+                        errors.Add(new EntityFailureException(failure.ErrorMessage));
+                    }
+                }
+            }
+
+            if (errors is not null)
+            {
+                throw errors.Count == 1 ? errors[0] : new AggregateException(errors);
+            }
+        }
```

Comment:
this method throws me off. It's called "Ensure success" but it collects failures. Can you tell me more about this? Maybe add some comments.

Reply:
Inspired by https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpresponsemessage.ensuresuccessstatuscode?view=net-7.0. It "ensures success" because if it is successfull it will _not_ throw.

I'll switch to `ThrowIfFailed` to be clearer